olive/passes/onnx/optimum_conversion.py

```diff
@@ -24,11 +27,41 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
         config = {
             "target_opset": PassConfigParam(
                 type_=int, default_value=14, description="The version of the default (ai.onnx) opset to target."
-            )
+            ),
+            "fp16": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Whether to use fp16 precision to load torch model and then convert it to onnx.",
+            ),
+            "device": PassConfigParam(
+                type_=str, default_value="cpu", description="The device to use to do the export. Defaults to 'cpu'."
+            ),
+            "no_post_process": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Whether to skip post-processing the exported model.",
+            ),
+            "extra_args": PassConfigParam(
+                type_=dict,
+                default_value=None,
+                description="Extra arguments to pass to the `optimum.exporters.onnx.main_export` function.",
+            ),
         }
         config.update(get_external_data_config())
         return config
 
+    def validate_search_point(
+        self, search_point: Dict[str, Any], accelerator_spec: AcceleratorSpec, with_fixed_value: bool = False
+    ) -> bool:
+        if with_fixed_value:
+            search_point = self.config_at_search_point(search_point or {})
+
+        if search_point.get("device") == "cuda" and not search_point.get("fp16"):
```

Comment:
I was thinking of the other direction, `if search_point.get("fp16") and "cuda" not in search_point.get("device")`. (or device=="cpu")
Similar to https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/conversion.py#L92.

Cuda is required for fp16 but not the other way around. We can still do fp32 on cuda.

Reply:
Oh, yes. Updated now! Thanks!

Reply:
`search_point.get("device") == "cpu" and search_point.get("fp16")` is cleaner. optimum does the same https://github.com/huggingface/optimum/blob/main/optimum/exporters/onnx/__main__.py#L290

Reply:
Oh just saw your update. I think your check is fine too. Unless we want to support something like `"cuda:0"`

Reply:
oh, cool. BTW, I previously try to add device id into Olive, but close that PR. There is a simple way to achieve that:
`CUDA_VISIBLE_DEVICES=1,2 python -m olive.workflows.run --config xxx`